src/components/[guild]/JoinModal/hooks/useActiveMembershipUpdate.tsx

```diff
@@ -69,7 +80,8 @@ const useActiveMembershipUpdate = ({
                 prev?.joinedAt || res?.done ? new Date().toISOString() : null,
               roles: [
                 ...(prev?.roles?.filter(
-                  (role) => !Object.keys(byRoleId).includes(role.roleId.toString())
+                  (role) =>
+                    !Object.keys(reqJobsByRoleId).includes(role.roleId.toString())
```

Comment:
Not related to this PR, I was just wondering if doing `!(role.roleId.toString() in reqJobsByRoleId)` would be the same without having to construct the keys array, and having to iterate over it

Reply:
According to a quick TS playground test, it would be the same, so changed it in https://github.com/guildxyz/guild.xyz/pull/1147/commits/ca21d263e135adf27a34243f8163c2a04a92a464 ;)